ipa-core/src/query/processor.rs

```diff
@@ -169,44 +175,51 @@ impl Processor {
             config: req,
             roles: roles.clone(),
         };
-
-        // Inform other parties about new query. If any of them rejects it, this join will fail
+        // Inform other helpers about new query. If any of them rejects it, this join will fail
         try_join(
             transport.send(left, prepare_request.clone(), stream::empty()),
             transport.send(right, prepare_request.clone(), stream::empty()),
         )
         .await
         .map_err(NewQueryError::MpcTransport)?;
 
+        shard_transport.broadcast(prepare_request.clone()).await?;
```

Comment:
I think we should first align our own shards and confirm that they are ready to execute before talking to other helpers. This code does not have a path to recover or a path to inform other helpers that query cannot move forward because one or more shards failed to confirm the query request